hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipUpdateTest.java

```diff
@@ -573,6 +574,21 @@ public void run() throws Exception {
         });
     }
 
+    @Test
+    public void shouldNotProcessStaleJoinRequest() {
+        HazelcastInstance hz1 = factory.newHazelcastInstance();
+        HazelcastInstance hz2 = factory.newHazelcastInstance();
+
+        JoinRequest staleJoinReq = getNode(hz2).createJoinRequest(true);
+        hz2.shutdown();
+        assertClusterSize(1, hz1);
```

Comment:
do we need eventual assertion? once hz2 completes shutdown, don't we guarantee that master node removes it from the member list.